homeassistant/components/tuya/event.py

```diff
@@ -0,0 +1,160 @@
+            translation_key="numbered_button",
+            translation_placeholders={"button_number": "8"},
+        ),
+        EventEntityDescription(
+            key=DPCode.SWITCH_MODE9,
+            device_class=EventDeviceClass.BUTTON,
+            translation_key="numbered_button",
+            translation_placeholders={"button_number": "9"},
+        ),
+    )
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: TuyaConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Tuya events dynamically through Tuya discovery."""
+    hass_data = entry.runtime_data
+
+    @callback
+    def async_discover_device(device_ids: list[str]) -> None:
+        """Discover and add a discovered Tuya binary sensor."""
+        entities: list[TuyaEventEntity] = []
+        for device_id in device_ids:
+            device = hass_data.manager.device_map[device_id]
+            if descriptions := EVENTS.get(device.category):
+                for description in descriptions:
+                    dpcode = description.key
+                    if dpcode in device.status:
+                        entities.append(
+                            TuyaEventEntity(device, hass_data.manager, description)
+                        )
+
+        async_add_entities(entities)
+
+    async_discover_device([*hass_data.manager.device_map])
+
+    entry.async_on_unload(
+        async_dispatcher_connect(hass, TUYA_DISCOVERY_NEW, async_discover_device)
+    )
+
+
+class TuyaEventEntity(TuyaEntity, EventEntity):
+    """Tuya Event Entity."""
+
+    entity_description: EventEntityDescription
+
+    def __init__(
+        self,
+        device: CustomerDevice,
+        device_manager: Manager,
+        description: EventEntityDescription,
+    ) -> None:
+        """Init Tuya event entity."""
+        super().__init__(device, device_manager)
+        self.entity_description = description
+        self._attr_unique_id = f"{super().unique_id}.{description.key}"
+
+        status_range = getattr(device, "status_range")
+
+        if (
+            description.key in status_range
+            and status_range[description.key].type == DPType.ENUM
+            and (
+                enum_type := EnumTypeData.from_json(
+                    DPCode(description.key), status_range[description.key].values
+                )
+            )
+        ):
+            self._attr_event_types: list[str] = enum_type.range
+
+    async def _handle_state_update(
+        self, updated_status_properties: list[str] | None
+    ) -> None:
+        if (
+            updated_status_properties is None
+            or self.entity_description.key not in updated_status_properties
+        ):
+            return
+
+        value = self.device.status.get(self.entity_description.key)
+        self._trigger_event(value)
+        self.async_write_ha_state()
+        if TYPE_CHECKING:
+            assert self.device_entry
+            assert self.registry_entry
```

Comment:
Not sure why this is added in the end?

```suggestion
```

Reply:
I also can't remember where this comes from. Removed it